redwood-protocol-compose/src/commonMain/kotlin/app/cash/redwood/protocol/compose/IdAllocator.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.protocol.compose
+
+import app.cash.redwood.protocol.Id
+
+/**
+ * @suppress For generated code usage only.
+ */
+public class IdAllocator {
```

Comment:
This type will get a new name and a lot more responsibility over the coming PRs. It becomes the Compose-side version of `ProtocolDisplay`.

Reply:
Will this eventually get the responsibility of converting Compose changes into diffs? Or will that still live in `ProtocolApplier`?